cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -325,11 +325,11 @@ long VpiSignalObjHdl::get_signal_value_long()
 }
 
 // Value related functions
-int VpiSignalObjHdl::set_signal_value(long value, gpi_set_action_t action)
+int VpiSignalObjHdl::set_signal_value(int64_t value, gpi_set_action_t action)
 {
     s_vpi_value value_s;
 
-    value_s.value.integer = (int)value;
+    value_s.value.integer = static_cast<PLI_INT32>(value);
```

Comment:
A little worried about silent truncation here.

Reply:
Actually, this should not be a problem as we are only expecting 32-bit values here.
I'm working on adding some proper value range checking in C++, which will ensure that we never overflow silently. However, I think this is more related to #268 rather than #1202, so unless you prefer another commit in this PR, I will address this in a separate PR together with the change to handle.py as suggested by @k tbarrett in the comment above.